src/js/libgeo/Prover.js

```diff
@@ -68,21 +74,77 @@ function incidentPC(p, c) {
             var erg = General.mult(c.matrix, p.homog);
             erg = General.mult(p.homog, erg);
             erg = CSNumber.abs(erg);
-            if (erg.value.real < 0.0000000000001) {
-                p.incidences.push(c.name);
-                c.incidences.push(p.name);
-            }
+            return (erg.value.real < 0.0000000000001);
         }
     };
 }
 
 function checkConjectures() {
+    var debug = false;
+    if (debug) console.log("conjectures", conjectures.length);
+    //if (!debug)
     if (conjectures.length === 0) return;
-    // TODO: we need some randomized proving here:
-    // move free elements and check conjectures a number of times.
-    // For now assume that all conjectures could be verified.
+    backupGeo();
+
+    var ii, jj, emove;
+    var nummoves = 3;
+
+    // filter free objects which are involved in conjectures
+    var involved;
+
+    var recalcInvolved = function() {
+        involved = [];
+        conjectures.forEach(function(con) {
+            var invs = con.getInvolved();
+            invs = invs.forEach(function(el) {
+                if (involved.indexOf(el) < 0 && el.movable) involved.push(el);
+            });
+        });
+    };
+
+    recalcInvolved();
+
+    // for jshint move the function definition outside loop 
+    var checkCon = function(con) {
+        return con.holds();
+    };
+
+    // debug code remove later
+    var nconject = conjectures.length;
+    for (var kk = 0; kk < nummoves; kk++) {
+        for (var oo = 0; oo < involved.length; oo++) {
+            var el = involved[oo];
+            if (!el.pinned) {
+                if (debug) console.log("prover: moving element", el.name);
+                // get random move and move free element
+                emove = geoOps[el.type].getRandomMove(el);
+                movepointscr(el, emove.value, emove.type);
+                // if something bad happens
+                if (tracingFailed) stateIn.set(stateLastGood);
```

Comment:
Do we want to set `stateLastGood` if tracing succeeded? Or in other words, if we entered a singularity, do we want to start from the original configuration or from the last state outside the singularity? The former has lower risk of entering the singularity again, but it may stay closer to the original (and possibly special) configuration. I'm not sure which to prefer. Currently you always start from the original, and that's fine for now. But I'd invite everyone to think about this distinction.